src/utils.ts

```diff
@@ -99,3 +100,19 @@ export function isGhes(): boolean {
   );
   return ghUrl.hostname.toUpperCase() !== 'GITHUB.COM';
 }
+
+export function isCacheFeatureAvailable(): boolean {
+  if (!cache.isFeatureAvailable()) {
+    if (isGhes()) {
+      throw new Error(
+        'Caching is only supported on GHES version >= 3.5. If you are on version >=3.5 Please check with GHES admin if Actions cache service is enabled or not.'
+      );
```

Comment:
Since caching is just an optimization, I feel like a warning would be more appropriate here.

Reply:
we were already throwing an error in the case of GHES https://github.com/actions/setup-python/blob/main/src/setup-python.ts#L15. So i didn't change the end-user expectations. Should we change this behaviour?

Reply:
Agree, it will be a breaking behavior. In `actions/cache` action we do log a warning only. But here it was treated as an error earlier, may be because caching is additionally asked by user (by setting `cache` input) during nodejs setup.

Reply:
Ok, I can see this as an error since it's in line with the previous behavior.  In that case, just a small edit to the text:

```suggestion
      throw new Error(
        'Caching is only supported on GHES version >= 3.5.  If you are on a version >= 3.5, please check with your GHES admin if the Actions cache service is enabled or not.'
      );
```

Although, I'm confused here: can the Actions cache service be enabled/disabled independently?

Reply:
yeah as the user is explicitly asking for cache, will it make more sense to throw an error in case the cache service is not there.  I can see both pros and cons for throwing errors and warnings. I am not sure of the initial thought process behind throwing an error instead of a warning(i believe mostly because a user has asked explicitly for caching).  Please let me know your thoughts? I am open to both